ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/VulnReportingLayout.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { PageSection, Title, Tabs, Tab } from '@patternfly/react-core';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
+
+import useFeatureFlags from 'hooks/useFeatureFlags';
+import PageTitle from 'Components/PageTitle';
+import {
+    vulnerabilityConfigurationReportsPath,
+    vulnerabilityOnDemandReportsPath,
+} from 'routePaths';
+
+const tabs = [
+    {
+        id: 'report-configuration',
+        title: 'Report configurations',
+        path: vulnerabilityConfigurationReportsPath,
+    },
+    { id: 'on-demand-reports', title: 'On-demand reports', path: vulnerabilityOnDemandReportsPath },
+];
+
+function VulnReportingLayout() {
+    const { isFeatureFlagEnabled } = useFeatureFlags();
+    const isOnDemandReportsEnabled = isFeatureFlagEnabled('ROX_VULNERABILITY_ON_DEMAND_REPORTS');
+
+    const location = useLocation();
+    const navigate = useNavigate();
+
+    const activeTabIndex = tabs.findIndex((tab) => location.pathname.startsWith(tab.path));
+
+    const onTabSelect = (_event, tabIndex) => {
+        navigate(tabs[tabIndex].path);
+    };
+
+    return (
+        <>
+            <PageTitle title="Vulnerability reporting" />
+            <PageSection
+                variant="light"
+                className={`${!isOnDemandReportsEnabled && 'pf-v5-u-pb-0'}`}
+            >
+                <Title headingLevel="h1">Vulnerability reporting</Title>
+            </PageSection>
+            {isOnDemandReportsEnabled && (
+                <PageSection
+                    variant="light"
+                    padding={{ default: 'noPadding' }}
+                    className="pf-v5-u-pl-lg pf-v5-u-background-color-100"
+                >
+                    <Tabs activeKey={activeTabIndex} onSelect={onTabSelect}>
+                        {tabs.map((tab, index) => (
+                            <Tab
+                                key={tab.id}
+                                eventKey={index}
+                                title={tab.title}
+                                tabContentId={`${tab.id}-tab-content`}
+                            />
+                        ))}
+                    </Tabs>
+                </PageSection>
+            )}
+            <PageSection padding={{ default: 'noPadding' }}>
+                <Outlet />
```

Comment:
Not a request for change, and it is possibly just due to my lack of experience using `<Outlet />`, but I have a gut reaction to reject this pattern. The fact that a component has this prop-less self-closing component in it, that renders some large amount of content invisible to the current component doesn't sit well with me.

I'll have to read the docs on this more and see if I can make peace with the react-router API. 🤔

Reply:
This might be a good way! Just something I don't have a ton of experience with recently and at first glance it looks very implicit in what is going to be rendered there. 😅 

The fact that the component is called `_Layout` though makes this more clear what pattern is in use.